Rules/UseUsingScopeModifierInNewRunspaces.cs

```diff
@@ -0,0 +1,493 @@
+                        commandAst.CommandElements.IndexOf(scriptBlockExpressionAst) - 1] as CommandParameterAst;
+
+                if (IsInlineScriptBlock(cmdName) || 
+                    IsJobScriptBlock(cmdName, scriptBlockParameterAst) ||
+                    IsForeachScriptBlock(cmdName, scriptBlockParameterAst) ||
+                    IsInvokeCommandComputerScriptBlock(cmdName, commandAst) ||
+                    IsDSCScriptResource(cmdName, commandAst))
+                {
+                    AnalyzeScriptBlock(scriptBlockExpressionAst);
+                    return AstVisitAction.SkipChildren;
+                }
+
+                if (IsInvokeCommandSessionScriptBlock(cmdName, commandAst))
+                {
+                    if (!TryGetSessionNameFromInvokeCommand(commandAst, out var sessionName))
+                    {
+                        return AstVisitAction.Continue;
+                    }
+
+                    IEnumerable<VariableExpressionAst> varsInLocalAssignments = FindVarsInAssignmentAsts(scriptBlockExpressionAst);
+                    if (varsInLocalAssignments != null)
+                    {
+                        AddAssignedVarsToSession(sessionName, varsInLocalAssignments);
+                    }
+
+                    GenerateDiagnosticRecords(
+                        FindNonAssignedNonUsingVarAsts(
+                            scriptBlockExpressionAst, 
+                            GetAssignedVarsInSession(sessionName)));
+
+                    return AstVisitAction.SkipChildren;
+                }
+                
+                return AstVisitAction.Continue;
+            }
+
+            /// <summary>
+            /// FindVarsInAssignmentAsts: Retrieves all assigned variables from an Ast:
+            /// Example: `$foo = "foo"` ==> the VariableExpressionAst for $foo is returned
+            /// </summary>
+            /// <param name="ast"></param>
+            private static IEnumerable<VariableExpressionAst> FindVarsInAssignmentAsts(Ast ast)
+            {
+                // Find all variables that are assigned within this ast
+                foreach (Ast foundAst in ast.FindAll(VarsInAssignments, true))
+                {
+                    var assignment = foundAst as AssignmentStatementAst;
+                    
+                    if (assignment.Left is VariableExpressionAst variable)
+                    {
+                        yield return variable;
+                    }
+                    else if (assignment.Left is ConvertExpressionAst conversion)
+                    {
+                        yield return conversion.Child as VariableExpressionAst;
+                    }
+                };
+            }
+
+            /// <summary>
+            /// VarsInAssignments: helper function to prevent allocation of closures for FindAll predicate.
+            /// </summary>
+            /// <param name="ast"></param>
+            private static bool VarsInAssignments(Ast ast)

```

Comment:
A predicate method should have a boolean name that starts with a verb, like `IsAssignmentStatement`